### ISSUE TITLE
flutter_markdown accessibility issues with VoiceOver on MacOS

### ISSUE THREAD

**Issue by hapasa** [flutter/flutter#162235]
### Steps to reproduce

When VoiceOver is on in MacOS, in a single bullet list I only hear "bullet", not the text itself.  That is the bug main bug.

 

### Expected results

* this is a test

Should announce something like "bullet, this is a text."   now I only hear "bullet"

But it should also be noted that navigating markdown doc with VoiceOver is quite cumbersome.  I tried wrapping Markdown widget to MergeSemantics, and that did not help at all.  

### Actual results

I only hear "bullet" announced by VO for single bullet lists.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter_markdown/flutter_markdown.dart';

void main() {
  runApp(const MyApp());
}

class ChangelogViewer extends StatelessWidget {
  final String assetPath;

  const ChangelogViewer({
    super.key,
    this.assetPath = 'assets/changelog.md',
  });

  @override
  Widget build(BuildContext context) {
    return FutureBuilder(
      future: DefaultAssetBundle.of(context).loadString(assetPath),
      builder: (BuildContext context, AsyncSnapshot<String> snapshot) {
        if (snapshot.hasData) {
          return Markdown(
            data: snapshot.data!,
            selectable: true,
            padding: const EdgeInsets.all(16),
          );
        } else if (snapshot.hasError) {
          return Center(
            child: Text(
              'Error loading changelog: ${snapshot.error}',
              style: TextStyle(color: Theme.of(context).colorScheme.error),
            ),
          );
        } else {
          return const Center(
            child: CircularProgressIndicator(),
          );
        }
      },
    );
  }
}


class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Changelog'),
        ),
        body: const ChangelogViewer(),
      ),
    );
  }
}

```

changelog.md:
```
* this text is not VO accessible

Some generic text

1. Now accessing this list
1. is like an escape room
```

pubspec.yaml, add:
```
  assets:
     - assets/changelog.md

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.27.3, on macOS 15.2 24C101 darwin-arm64, locale en-FI)
    • Flutter version 3.27.3 on channel stable at /Users/harri/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c519ee916e (5 days ago), 2025-01-21 10:32:23 -0800
    • Engine revision e672b006cb
    • Dart version 3.6.1
    • DevTools version 2.40.2

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/harri/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16C5032a
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.59.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 15.2 24C101 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 15.2 24C101 darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome 132.0.6834.110

[✓] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by darshankawar** [comment]
Thanks for the report. I was able to replicate this based on the details provided.

```
stable : 3.27.3
master : 3.29.0-1.0.pre.127
```
